src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -317,6 +317,16 @@ private void RunThread(int timeout)
 
             if (!thread.Join(timeout))
             {
+                // Don't enforce timeout when debugger is attached.
+                // We perform this check after the initial timeout has passed to 
+                // give the user additional time to attach a debugger 
+                // after the test has started execution
+                if (Debugger.IsAttached)
```

Comment:
Interesting. I wouldn't have thought to do it quite this way but I guess it allows for a very last minute attaching of the process.